runner/bsp/worker/src/mill/bsp/worker/BspEvaluators.scala

```diff
@@ -14,30 +15,74 @@ private[mill] class BspEvaluators(
     val watched: Seq[Watchable]
 ) {
 
+  private val transitiveDependencyModules0 = new ConcurrentHashMap[ModuleApi, Seq[ModuleApi]]
+  private val transitiveModulesEnableBsp0 =
+    new ConcurrentHashMap[ModuleApi, Option[Seq[BspModuleApi]]]
```

Comment:
@alexarchambault can you explain the changes here? I would have expected them to be a simple `.filter(_.enableBsp)`, not sure what the `transitiveDependencyModules` stuff is. If it's to resolve a different issue, it should be described in the PR description